src/front/glsl/helpers.rs

```diff
@@ -235,6 +236,87 @@ pub fn glsl_to_spirv_type(ty: TypeSpecifierNonArray) -> Option<TypeInner> {
             kind: ScalarKind::Float,
             width: 64,
         },
+        TypeName(ty_name) => {
+            if ty_name.0.contains("texture") {
+                let (base, offset) = match &ty_name.0[..1] {
+                    "t" => {

```

Comment:
nit: looks like you only need an `&str` and `ScalarKind` out of the match and do all the code after it